libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -478,7 +613,7 @@ int HTTPClient::sendRequest(const char * type, uint8_t * payload, size_t size)
 
     // send Payload if needed
     if(payload && size > 0) {
-        if(_tcp->write(&payload[0], size) != size) {
+        if(_client->write(&payload[0], size) != size) {
```

Comment:
I'm definitely not a fan of the `#ifdef` bits, but in prior instances in this file I can understand your logic.  You had to have both `_tcpDeprecated` and `_client` operations in the code, with appropriate `if ()`s to ensure the right one was used.  You were kind of stuck with doing both cases due to the way it was being implemented having 2 potential `WiFiClient`s to use.

But don't you need the `if (tcp) tcp->write(); else _client->write();` pattern here and elsewhere below in the code where you `s/_tcp/_client/g`'d?  The way I see it now, old code will simply never write or read anymore, since `_tcpDeprecated` is not being read/written.